test-tools/wamr-ide/VSCode-Extension/src/utilities/directoryUtilities.ts

```diff
@@ -118,3 +120,44 @@ export function checkFolderName(folderName: string) {
 
     return valid;
 }
+
+export function DownloadFile(url: string, destinationPath: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+        const file = fileSystem.createWriteStream(destinationPath);
+        const stream = request(url).pipe(file);
+        stream.on("close", resolve);
+        stream.on("error", reject);
+    });
+}
+
+export function UnzipFile(sourcePath: string, getDestinationFileName: (entryName: string) => string): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+        const unzippedFilePaths: string[] = [];
+        yauzl.open(sourcePath, { lazyEntries: true }, function(error, zipfile) {
+            if (error) reject(error);
```

Comment:
don't you need to return at this point? Otherwise the execution will flow to the next line.

Reply:
Good catch, thanks!